File: Old/Deck.py
'''
this is not actually the deck
it is all possible cards that we don't already know have be dealt out
'''

import random

class Deck:
    
    cards=[]
    
    def __init__(self):
        for suit in Card.suits:
            for faceValue in Card.faceValues:
                self.cards.append(Card(faceValue, suit))

                
    def dealRandomCard(self):
        card=random.sample(self.cards,  1)[0]
        self.cards.remove(card)
        return card

    #card must be in the deck    
    def dealCard(self, card):
        self.cards.remove(card)
        return card
    
    #card must not be in the deck
    def putBack(self, Card):
        self.cards.append(Card)
        
    def getCards(self):
        return self.cards
      
      
class Card:
    suits = ['h','c','s','d']
    faceValues = ['2','3','4','5','6','7','8','9','T','J','Q','K','A']
    
    def __init__(self, faceValue, suit):
        self.faceValue=faceValue
        self.suit = suit
    def __str__(self):
        return self.faceValue + self.suit
    def __eq__(self,other):
        return self.faceValue==other.faceValue and self.suit==other.suit

File: Java Bot/src/cards/Deck.java
package cards;

import java.util.*;

import cards.Card.Rank;
import cards.Card.Suit;

public class Deck {
    
    private final List<Card> cards;
    private final Random generator;
    private static final Rank ranks[] = {Rank.TWO,Rank.THREE,Rank.FOUR,Rank.FIVE,Rank.SIX,Rank.SEVEN,
                                Rank.EIGHT,Rank.NINE,Rank.TEN,Rank.JACK,Rank.QUEEN,Rank.KING,Rank.ACE};
    private static final Suit suits [] = {Suit.CLUB,Suit.DIAMOND,Suit.SPADE,Suit.HEART};
    
    public Deck(){
        generator = new Random();
        
        cards = new ArrayList<Card>();
        for(Suit suit :suits){
            for(Rank rank: ranks){
                cards.add(new Card(rank,suit));
            }
        }
    }
    
    
    /**
     * @return a random card from the deck. Removes the card from the deck
     */
    public Card dealCard(){
        int index = generator.nextInt(cards.size());
        return cards.remove(index);
    }
    
    
    /**
     * @param card card to remove from the deck
     * @return true if the card was removed, false otherwise
     */
    public Boolean dealCard(Card card){
        return cards.remove(card);
    }
    
    
    /**
     * @param card card to be put back into the deck. Card must not already be in the deck
     */
    public void putBack(Card card){
        cards.add(card);
    }
    
    
    /**
     * @return all of the cards in this deck
     */
    public List<Card> getCards(){
        return new ArrayList<Card>(cards);
    }
}
